packages/core/services/fetchComponents.js

```diff
@@ -23,9 +23,24 @@ export async function fetchComponents(routeMeta, routeCookies) {
     hostname,
   } = routeMeta;
   const env = getEnv(hostname);
-  const { FETCH_TIMEOUT } = env;
+  const { FETCH_TIMEOUT, API_ROOT_URL } = env;
   const apiUrl = createEndpointUrl(routeMeta, routeCookies);
 
+  /**
+   * If for some reason the API_ROOT_URL is not defined or a user/bot finds
+   * an URL that is not mapped to a site on the network, return early
+   * and avoid making and invalid fetch request ex:(https://undefined/foo/bar).
+   */
+  if (!API_ROOT_URL) {
+    return {
```

Comment:
I think you could include a `log.warn` here if you wanted too. It would only cause noise locally, and even then probably only rarely (or someone could just filter out those logs)